lib/WebpackError.js

```diff
@@ -4,8 +4,81 @@
 */
 "use strict";
 
+const createHash = require("./util/createHash");
+
+const hash = Symbol();
+const meta = Symbol();
+
 module.exports = class WebpackError extends Error {
+	constructor(
+		message,
+		line = 0,
+		column = 0,
+		columnEnd = 0,
+		category = "error"
```

Comment:
Is this redundant information? It's already stated by the name of the array to which the error is pushed to.

Can I push an error to the `compilation.warnings` array?

Reply:
It is not. Category will be used in further PRs, which satisfies a portion of the RFC allowing errors to use a name/category for identifying what the error is for. For example, configuration errors would not include "configuration" as a prefix, rather they would set the category to "configuration" and keep the error message property short. "error", "warning", "configuration", "performance" are all examples of categories. This will be utilized more when the abstract enforcement goes into place.